minari/data_collector/data_collector.py

```diff
@@ -348,76 +348,60 @@ def clear_buffer(dictionary_buffer: EpisodeBuffer, episode_group: h5py.Group):
             """
             for key, data in dictionary_buffer.items():
                 if isinstance(data, dict):
-
-                    if key in episode_group:
-                        eps_group_to_clear = episode_group[key]
-                    else:
-                        eps_group_to_clear = episode_group.create_group(key)
+                    eps_group_to_clear = episode_group.get(
+                        key, episode_group.create_group(key)
+                    )
                     clear_buffer(data, eps_group_to_clear)
-                elif all([isinstance(entry, tuple) for entry in data]):
+                elif all(map(lambda elem: isinstance(elem, tuple), data)):
                     # we have a list of tuples, so we need to act appropriately
                     dict_data = {
                         f"_index_{str(i)}": [entry[i] for entry in data]
                         for i, _ in enumerate(data[0])
                     }
-                    if key in episode_group:
-                        eps_group_to_clear = episode_group[key]
-                    else:
-                        eps_group_to_clear = episode_group.create_group(key)
+                    eps_group_to_clear = episode_group.get(
+                        key, episode_group.create_group(key)
+                    )
                     clear_buffer(dict_data, eps_group_to_clear)
-                elif all([isinstance(entry, OrderedDict) for entry in data]):
-
+                elif all(map(lambda elem: isinstance(elem, OrderedDict), data)):
                     # we have a list of OrderedDicts, so we need to act appropriately
                     dict_data = {
                         key: [entry[key] for entry in data]
                         for key, value in data[0].items()
                     }
-
-                    if key in episode_group:
-                        eps_group_to_clear = episode_group[key]
-                    else:
-                        eps_group_to_clear = episode_group.create_group(key)
+                    eps_group_to_clear = episode_group.get(
+                        key, episode_group.create_group(key)
+                    )
                     clear_buffer(dict_data, eps_group_to_clear)
                 else:
-                    # convert data to numpy
-                    np_data = np.asarray(data)
-                    assert np.all(
-                        np.logical_not(np.isnan(np_data))
-                    ), "Nan found after cast to nump array, check the type of 'data'."
+                    if all(map(lambda elem: isinstance(elem, str), data)):
+                        data_shape = (len(data),)
+                        dtype = h5py.string_dtype(encoding="utf-8")
```

Comment:
Oh I see how this is a bottleneck as there's only this option or ascii (which is less chars), so given that I think it's fine to do utf-8 https://docs.h5py.org/en/stable/special.html#h5py.string_dtype

Reply:
Did some more searching around and this shouldn't be a problem after all, you can still encode full unicode characters using utf-8 (on unix they are encoded that way by default, on windows apparently it's utf-16), it just uses multi-byte encoding (https://unicodebook.readthedocs.io/unicode_encodings.html). Not sure how exactly it would work with gymnasium spaces but I think it should be good enough, utf-8 is the standard for pretty much everything. If a user really needs to use unicode characters in their gymnasium action/obs space I think they should be able to set the charset as just utf-8 and encode the unicode characters into utf-8.